src/python/pants/backend/native/subsystems/native_toolchain.py

```diff
@@ -186,20 +193,19 @@ def select_llvm_cpp_toolchain(platform, native_toolchain):
       # NB: we use g++'s headers on Linux, and therefore their C++ standard library.
       include_dirs=provided_gpp.include_dirs,
       extra_args=(llvm_cpp_compiler_args + provided_clangpp.extra_args + gcc_install.as_clang_argv))
-    linking_library_dirs = provided_gpp.library_dirs + provided_clangpp.library_dirs
+    # TODO: why are these necessary? this is very mysterious.
```

Comment:
+1 that we should better understand why certain toolchains require knowing about both library dirs or both include dirs.

Reply:
I don't know if it is required -- that's why this PR is so necessary (testing all of these configurations in an integration test).